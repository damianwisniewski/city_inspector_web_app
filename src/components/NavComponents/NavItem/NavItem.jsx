import React from 'react'
import PropTypes from 'prop-types'
import { NavLink } from 'react-router-dom'

import './NavItem.scss'
import iconAssets from '../../../assets/styleModules/icons.module.scss'
import { PopupLink } from '../../PopupComponents/PopupRouting'

const NavItem = ({ id, children, type, className, icon, exact, to }) => {
	function createProperChild() {
		if (type === 'link') {
			return (
				<NavLink
					id={id}
					type={type}
					exact={exact}
					to={to}
					activeClassName='navigation_active'
					className={`nav_link ${iconAssets[icon]} ${className} `}
				>
					{children}
				</NavLink>
			)
		} else {
			return (
				<PopupLink
					id={id}
					type={type}
					to={to}
					activeClassName='navigation_active'
					className={`nav_button ${iconAssets[icon]} ${className} `}
				>
					{children}
				</PopupLink>
			)
		}
	}

	return <li className='nav_item'>{createProperChild()}</li>
}

NavItem.defaultProps = {
	children: '',
	icon: '',
	exact: false,
	className: '',
}

NavItem.propTypes = {
	id: PropTypes.string,
	children: PropTypes.node,
	icon: PropTypes.string,
	exact: PropTypes.bool,
	className: PropTypes.string,
	type: PropTypes.oneOf(['link', 'button']).isRequired,
	to: PropTypes.string.isRequired,
}

export default NavItem
