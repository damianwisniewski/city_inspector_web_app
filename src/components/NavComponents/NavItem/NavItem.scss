@import '../../../assets/styleHelpers/index';

.nav_item {
	margin: 0 10px;
	text-align: center;

	@media only screen and (max-width: 950px) {
		background-color: $primary_light_blue;
		border-radius: 5px;
		margin: 10px;
	}

	.nav_button {
		width: 100%;

		border: 0;
		outline: 0;

		padding: 0;
		margin: 0;

		background-color: transparent;

		cursor: pointer;
	}

	.nav_link,
	.nav_button {
		font-family: $primary_font;
		font-size: 15px;
		text-decoration: none;
		color: $primary_light_blue;
		white-space: nowrap;

		transition: all 0.2s ease-in;

		@media only screen and (max-width: 950px) {
			display: flex;
			align-items: center;
			padding: 10px;
			color: $primary_dark_blue;

			&::before {
				margin-left: 0;
				margin-right: auto;
			}
		}

		&:not(.navigation_active):hover {
			color: $gray;
		}

		&.navigation_active {
			color: $white;

			&::before {
				color: $white;
			}
		}
	}
}
