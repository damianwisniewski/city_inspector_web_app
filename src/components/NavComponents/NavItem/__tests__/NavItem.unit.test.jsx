import React from 'react'
import NavItem from '../NavItem'
import { configure, shallow, ShallowWrapper } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

configure({ adapter: new Adapter() })

describe('<NavItem />', () => {
	/**
	 * @type {ShallowWrapper}
	 */
	let NavItemComponent
	const NavItemText = 'ExampleText'
	const location = '/'
	const icon = 'test'
	const linkType = 'link'
	const buttonType = 'button'

	test('should render correctly NavItem type BUTTOM', () => {
		NavItemComponent = shallow(
			<NavItem icon={icon} type={buttonType} to={location}>
				{NavItemText}
			</NavItem>,
		)
		expect(toJson(NavItemComponent)).toMatchSnapshot()
	})

	test('should render correctly NavItem type LINK', () => {
		NavItemComponent = shallow(
			<NavItem icon={icon} type={linkType} to={location}>
				{NavItemText}
			</NavItem>,
		)

		expect(toJson(NavItemComponent)).toMatchSnapshot()
	})
})
