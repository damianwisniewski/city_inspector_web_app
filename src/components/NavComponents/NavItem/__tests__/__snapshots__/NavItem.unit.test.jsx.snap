// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavItem /> should render correctly NavItem type BUTTOM 1`] = `
<li
  className="nav_item"
>
  <PopupLink
    activeClassName="navigation_active"
    className="nav_button test  "
    to="/"
    type="button"
  >
    ExampleText
  </PopupLink>
</li>
`;

exports[`<NavItem /> should render correctly NavItem type LINK 1`] = `
<li
  className="nav_item"
>
  <NavLink
    activeClassName="navigation_active"
    className="nav_link test  "
    exact={false}
    to="/"
    type="link"
  >
    ExampleText
  </NavLink>
</li>
`;
